### PR TITLE
Add support for kubernetes builds on the `from-token` deployments

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR adds support for kubernetes builds on the `from-token` deployments using `--kubernetes` flag as well as adds support for specifying whether the deployment should run directly or not using `--no-deploy` flag

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by 8ball030 — DISMISSED** [review]
Thanks

**Comment by 8ball030** [comment]
image

:)

File: autonomy/cli/helpers/deployment.py [hunk 1]
```diff
@@ -263,4 +265,8 @@ def build_and_deploy_from_token(  # pylint: disable=too-many-arguments, too-many
             build_image(agent=service.agent)
 
     click.echo("Service build successful.")
+    if no_deploy or deployment_type == KubernetesGenerator.deployment_type:
```
Comment (solarw) [inline autonomy/cli/helpers/deployment.py:268]: test this case with mocking  `run_deployment` not called if k8s or no deploy?

**Review by solarw — APPROVED** [review]
lgtm

**Pull request merged**